Modules/_threadmodule.c

```diff
@@ -143,19 +140,26 @@ ThreadHandle_get_os_handle(ThreadHandle *handle, PyThread_handle_t *os_handle)
 }
 
 static void
-add_to_shutdown_handles(thread_module_state *state, ThreadHandle *handle)
+add_to_shutdown_handles(struct llist_node *shutdown_handles,
+                        ThreadHandle *handle)
 {
     HEAD_LOCK(&_PyRuntime);
-    llist_insert_tail(&state->shutdown_handles, &handle->shutdown_node);
+    llist_insert_tail(shutdown_handles, &handle->shutdown_node);
     HEAD_UNLOCK(&_PyRuntime);
 }
 
-static void
-clear_shutdown_handles(thread_module_state *state)
+// Remove any remaining handles (e.g. if shutdown exited early due to
+// interrupt) so that attempts to unlink the handle after our module state
+// is destroyed do not crash.
+void
+_PyThread_ClearThreadHandles(PyInterpreterState *interp)
```

Comment:
Core functionality (e.g. API called in pylifecycle.c or pystate.c) should be implemented under the Python directory, rather than in one of the Modules files.  Let's move this function to Python/thread.c, where the rest of the `PyThread_*` API lives.  That would mean you don't have to add this file to the PCbuild files.

Reply:
Eric and I chatted in person. This would involve a larger change that moves `ThreadHandle` and associated functions. We want this to be a targeted fix, so that should happen in a follow on PR.